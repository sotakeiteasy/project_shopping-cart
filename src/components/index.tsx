import Header from "./Header/Header";
import ProductCard from "./ProductCard/ProductCard";
export { Header, ProductCard };
