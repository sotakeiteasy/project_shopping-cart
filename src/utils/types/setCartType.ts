import { Product } from "./product";
export type SetCartType = React.Dispatch<React.SetStateAction<Product[]>>;
