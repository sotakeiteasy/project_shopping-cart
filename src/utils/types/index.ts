import { Product } from "./product";
import { SetCartType } from "./setCartType";

export type { Product, SetCartType };
