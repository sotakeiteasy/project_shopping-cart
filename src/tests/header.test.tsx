import { render, screen } from "@testing-library/react";
import { Header } from "@/components";
import { Product } from "@/utils/types";
import { BrowserRouter } from "react-router";


describe("header renders", () => {
    const mockedCart: Product[] = []

    it('snapshot header', () => {
        const { container } = render(
            <BrowserRouter>
                <Header cart={mockedCart} />
            </BrowserRouter>
        )
            
        expect(container).toMatchSnapshot()
    })

    it('count shows length of cart', () => {
        const mockedCart: Product[] = [{
            id: 1,
            title: 'test',
            price: 20,
            description: "test",
            category: "test",
            image: "test",
            rating: { count: 2, rate: 5 },
        },
        {
            id: 1,
            title: 'test',
            price: 20,
            description: "test",
            category: "test",
            image: "test",
            rating: { count: 2, rate: 5 },
        }]

        render(
            <BrowserRouter>
                <Header cart={mockedCart} />
            </BrowserRouter>
        )
            
        expect(screen.getByText(/cart 2/i)).toBeInTheDocument()
    })
})
