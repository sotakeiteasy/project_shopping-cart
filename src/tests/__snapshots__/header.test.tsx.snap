// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`header renders > snapshot header 1`] = `
<div>
  <header
    class="_header_98091d"
  >
    <nav
      class="_nav_98091d"
    >
      <a
        class="_navLink_98091d _logo_98091d"
        data-discover="true"
        href="/"
      >
        My Store
      </a>
      <a
        class="_navLink_98091d"
        data-discover="true"
        href="/catalog"
      >
        Catalog
      </a>
      <div
        class="_navGroup_98091d"
      >
        <a
          class="_navLink_98091d"
          data-discover="true"
          href="/cart"
        >
          Cart 
          0
        </a>
      </div>
    </nav>
  </header>
</div>
`;
